library/swift/src/RetryPolicy.swift

```diff
@@ -3,7 +3,7 @@ import Foundation
 /// Rules that may be used with `RetryPolicy`.
 /// See the `x-envoy-retry-on` Envoy header for documentation.
 @objc
-public enum RetryRule: Int {
+public enum RetryRule: Int, CaseIterable {
```

Comment:
nit: I know it's not strictly part of this PR, but what about `case status5xx` and `case retriable4xx`?

I just wonder if the current naming might be a little less obvious to some.

Reply:
They're named this way at the moment to mirror upstream envoy (see the `stringValue` mappings below). We can rename them if you think this would make them clearer - do you have context into why they were named this way upstream?

Reply:
Right, I was actually thinking this maybe mapped closer to upstream (e.g. `retriable4xx` instead of `retriableFourXX`)?

I recognize `status5xx` is a little bit of a departure, but I think `5xx` is maybe more common/recognizable as a class of status codes than `fiveXX` might be.

No very strong feelings about it though.

Reply:
SGTM, updating. cc @buildbreaker for Kotlin

Reply:
Done